ptx/sec_taylor_poly.ptx

```diff
@@ -583,16 +583,16 @@
                 xmin=-1,xmax=11
                 ]
 
-                \addplot+ [rightarrow,domain=0:3.16,samples=40] ({x^2},{x}) node [pos=1,above left] { $y=ln(x)$};
-                \addplot+ [rightarrow] coordinates {(0,0.5469)(0.2,0.6536)(0.4,0.7551)(0.6,0.8518)(0.8,0.944)(1.,1.032)(1.
-                2,1.116)(1.4,1.196)(1.6,1.273)(1.8,1.346)(2.,1.417)(2.2,1.485)(2.4,1.
-                55)(2.6,1.613)(2.8,1.673)(3.,1.732)(3.2,1.789)(3.4,1.844)(3.6,1.897)(
-                3.8,1.949)(4.,2.)(4.2,2.049)(4.4,2.098)(4.6,2.145)(4.8,2.191)(5.,2.
-                236)(5.2,2.28)(5.4,2.324)(5.6,2.366)(5.8,2.408)(6.,2.448)(6.2,2.488)(
-                6.4,2.527)(6.6,2.565)(6.8,2.602)(7.,2.637)(7.2,2.672)(7.4,2.705)(7.6,
-                2.737)(7.8,2.768)(8.,2.797)(8.2,2.824)(8.4,2.849)(8.6,2.873)(8.8,2.
-                894)(9.,2.913)(9.2,2.929)(9.4,2.942)(9.6,2.953)(9.8,2.96)(10.,2.964)} node [pos=0.8,below right] { $y=p_{20}(x)$};
-
+                \addplot+ [rightarrow,domain=0.01:3,smooth,samples=50] {ln(x)} node [pos=1,above left] { $y=ln(x)$};
+                \addplot+ [rightarrow,domain=-4:4,smooth] coordinates {(-0.108,-7.567)(-0.052,-4.958)(0.004,-3.519)(0.06,-2.671)(0.116,-2.13)
+                (0.172,-1.756)(0.228,-1.478)(0.284,-1.259)(0.34,-1.079)(0.396,-0.9263)
+                (0.452,-0.7941)(0.508,-0.6773)(0.564,-0.5727)(0.62,-0.478)(0.676,-0.
+                3916)(0.732,-0.312)(0.788,-0.2383)(0.844,-0.1696)(0.9,-0.1054)(0.956,-
+                0.045)(1.012,0.01193)(1.068,0.06579)(1.124,0.1169)(1.18,0.1655)(1.236,
+                0.2119)(1.292,0.2562)(1.348,0.2986)(1.404,0.3393)(1.46,0.3784)(1.516,
+                0.4161)(1.572,0.4523)(1.628,0.4874)(1.684,0.5212)(1.74,0.5538)(1.796,
+                0.5853)(1.852,0.6154)(1.908,0.6427)(1.964,0.6635)(2.02,0.6665)(2.076,
+                0.621)(2.132,0.4476)(2.188,-0.04877)(2.244,-1.326)(2.3,-4.421)} node [pos=0.8,below right] { $y=p_{20}(x)$};
```

Comment:
Where possible, I like to replace these point lists with more meaningful code expressions. Here, like `-(x-1)^20/20+(x-1)^19/19±...`. For one thing, the error here would have been caught sooner.

Using `smooth` raises some flags. It's probably fine, but since it occasionally does bad things, I try to avoid it. Did you try just upping the `samples`?

Only if you are interested...

For `ln(x)`, plotting it as a function of x near 0 of course causes smoothness issues. In situations like that, using parametric plotting and transformations makes good results. Like, try this:
```
\addplot[domain=0.01:3] ({x},{ln(x)});
```
which is bad. But then try this:
```
\addplot[domain=-4.6:1.1] ({e^x},{x});   %y=ln(x)
```
or perhaps better:
```
\addplot[domain=-4.6:1.1,variable=t] ({e^t},{t});   %y=ln(x)
```
No need for `smooth`, or boosting `samples` past the default 20.